bindings/python/src/optional.hpp

```diff
@@ -6,19 +6,19 @@
 # define OPTIONAL_070108_HPP
 
 # include "boost_python.hpp"
-# include <boost/optional.hpp>
+# include <optional>
 
 template <class T>
 struct optional_to_python
 {
     optional_to_python()
     {
         boost::python::to_python_converter<
-            boost::optional<T>, optional_to_python<T>
+            std::optional<T>, optional_to_python<T>
         >();
     }
 
-    static PyObject* convert(boost::optional<T> const& x)
+    static PyObject* convert(std::optional<T> const& x)
```

Comment:
these functions, taking `optional` are part of the boost.python interface. I don't think this would work

Reply:
looking at other converters, it looks to me that they don't depend on boost types, also I expect to have some unit test failing if it doesn't work, but I'm not familiar with the python binding, can you help me to get this one right?

Reply:
I would also expect a unit test failing. and I think I was wrong, the `optional` here is the type being converted

Reply:
Ok, I will wait until appveyor is done and squash